Source/WebCore/page/Page.cpp

```diff
@@ -5042,9 +5042,12 @@ void Page::activeNowPlayingSessionUpdateTimerFired()
     chrome().client().hasActiveNowPlayingSessionChanged(hasActiveNowPlayingSession);
 }
 
-void Page::setLastAuthentication(LoginStatus::AuthenticationType authType)
+void Page::setLastAuthentication(LoginStatus::AuthenticationType authType, const String& username)
 {
-    auto loginStatus = LoginStatus::create(RegistrableDomain(mainFrameURL()), emptyString(), LoginStatus::CredentialTokenType::HTTPStateToken, authType, LoginStatus::TimeToLiveAuthentication);
+    String usernameToSet = emptyString();
+    if (!username.isEmpty())
+        usernameToSet = username;
+    auto loginStatus = LoginStatus::create(RegistrableDomain(mainFrameURL()), usernameToSet, LoginStatus::CredentialTokenType::HTTPStateToken, authType, LoginStatus::TimeToLiveAuthentication);
```

Comment:
I don't think we need the variable and if statement, could just be:
```suggestion
    auto loginStatus = LoginStatus::create(RegistrableDomain(mainFrameURL()), username, LoginStatus::CredentialTokenType::HTTPStateToken, authType, LoginStatus::TimeToLiveAuthentication);
```